### DISCUSSION TITLE
streams_generator_a2dp No sound

### DISCUSSION THREAD

**Discussion by wizwik** [pschatzmann/arduino-audio-tools#1489] [Q&A] [answered]
I am using the unmodified streams_generator_a2dp example on Arduino with a ESP32. I am using Arduino V2.3.2. It complies and connects ok. I am using a JBL Flip Essential 2 as a speaker which is a2dp.  I have tested the speaker with an iPhone and a Mac Pro.  I have tested the software with a number of other Bluetooth receiver such as headphones. The problem is I don't get any sound. I think I have missed a setting somewhere. Here is the connection message I receive W] A2DPStream.h : 285 - ==> state: Connected
A2DP is connected now...

**Comment by pschatzmann** [comment, marked as answer]
Not sure yet what's causing the [issue](https://github.com/pschatzmann/arduino-audio-tools/issues/1491), but in the meantime you can use the [basic example ](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-basic-api/base-generator-a2dp/base-generator-a2dp.ino)which should be preferred

**Reply by wizwik** [reply]
Hi Phil:

That worked! Thank you very much!

**Reply by wizwik** [reply]
I am not sure I said I was using a ESP32  Dev board. I am going to try to change frequencies, turn off the sound and set up tone bursts of a predetermined length..

**Comment by pschatzmann** [comment]
I committed a [correction](https://github.com/pschatzmann/arduino-audio-tools/commit/45839f17ebfa09687bcb45795e49eaa929d1e667) which should resolve the problem.
I think in your version it might work if you just change the startup logic cfg.startLogic = StartOnConnect; in the configuration.

**Reply by wizwik** [reply]
Hi Phil:
I get a compile error: 'struct audio_tools::AudioInfo' has no member named 'startLogic'
   cfg.startLogic = StartOnConnect;  // Phils correction
Do I need to update my version of the audio-tools?

**Reply by pschatzmann** [reply]
it should be cfgA2DP.startLogic = StartOnConnect; This is an attribute of [A2DPConfig](https://pschatzmann.github.io/arduino-audio-tools/classaudio__tools_1_1_a2_d_p_config.html)

**Reply by wizwik** [reply]
Hi Phil:  Thanks. It compiled ok. Regards

**Comment by wizwik** [comment]
Hi Phil:

Thanks for the update.

I will try it out.

Regards

John

**Reply by wizwik** [reply]
Hi Phil:

It ran ok. I turn on auto_reconnect and that made the connection timing easy.
Thanks

Regards

John

**Reply by wizwik** [reply]
Hi Phil:
I have a question regarding the use of xTaskCreatePinnedToCore. I have two threads that I run in my project. I found that I had to disable them to allow the Bluetooth connection to run. The two tasks have priority 1 and 2 and run in Core 1. I saw that the BlueToothA2Psource.h references a Task handle with #  define xTaskHandle TaskHandle_t so I suspect that there are calls to xTaskCreatePinnedToCore. So I wonder what the Priority is and if I set the Bluetooth task to have a higher priority. I have not found a way to set the Priority yet. Regards  John

**Reply by pschatzmann** [reply]
Why don't you just [look it up in the documentation](https://pschatzmann.github.io/ESP32-A2DP/html/class_bluetooth_a2_d_p_common.html#a68f9e168839f0faeb72705ccabbb6b7a) ?

**Reply by wizwik** [reply]
Correct. Found it. Thanks
